config/routes.rb

```diff
@@ -23,6 +23,9 @@
     end
   end
 
+  get "report", to: "reports#new", as: :report
+  post "report/update", to: "reports#update", as: :update_report
```

Comment:
I think i would rather do this:
```suggestion
  resource :report, only: [:show, :update]
```

It also implies you have to change the controller action name to `show`, and the update action should use the http verb `PUT`. So it's not a drop-in replacement, but i think it's worth it :)